src/main.rs

```diff
@@ -24,14 +28,27 @@ extern crate serde;
 extern crate serde_xml_rs;
 extern crate xml;
 
-pub fn main() {
-    let (mut project_loader, queries, _) = parse_args();
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let yaml = load_yaml!("cli.yml");
+    let matches = App::from(yaml).get_matches();
+
+    if let Some(ip_endpoint) = matches.value_of("endpoint") {
+        start_grpc_server_with_tokio(ip_endpoint)?;
+    } else {
+        start_using_cli(&matches);
+    }
```

Comment:
This will always be true because the endpoint has a default value in cli.yml, so we can never run in cli mode as far as i can tell.

Maybe use the fact that query does not have a default instead?:
```suggestion
    if let Some(query) = matches.value_of("query") {
        start_using_cli(&matches);
    } else {
        let ip_endpoint = matches.value_of("endpoint").unwrap();
        start_grpc_server_with_tokio(ip_endpoint)?;
    }
```